clap_complete/tests/testsuite/dynamic.rs

```diff
@@ -19,7 +19,7 @@ fn suggest_subcommand_subset() {
         clap_complete::dynamic::complete(&mut cmd, args, arg_index, current_dir).unwrap();
     let completions = completions
         .into_iter()
-        .map(|s| s.to_string_lossy().into_owned())
+        .map(|s| s.0.to_string_lossy().into_owned())
```

Comment:
Can you add shell-agnostic tests for the various cases?

Reply:
@epage would you approve introducing a completion assert function in the dynamic test file: something like https://github.com/ModProg/clap/blob/d22db5c32701dc33cdc20584a2f779973124bcb5/clap_complete/tests/testsuite/dynamic.rs#L7-L28

Reply:
We will eventually want to test
- Completing values besides the last
- Completing within values (not really supported atm)

Reply:
well, that could be supported by such a function, or macro.

The input could be

```
assert_complete(&mut cmd, "arg1 arg2 | after_cursor", ["expected", "completions"]);

When no `|` is in the input, the completions after the last arg is produced. When we support in token completions, the `|` could be placed inside.
```

Reply:
`|` or `[TAB]` can work.

For the "expected", we have two routes to go
- Pass in a struct of completion information so we precisely test what was returned (this could be the same struct we eventually switch` complete` to returning)
- Format the completions into a string, much like [completest does for nushell](https://github.com/assert-rs/completest/blob/main/src/nu.rs#L104)